drivers/i3c/i3c_i2c_virtual_controller.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Google LLC
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ * Copyright 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT virtual_i3c_i2c_controller
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(i3c_i2c_virtual_controller, CONFIG_I3C_LOG_LEVEL);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i3c.h>
+
+struct i2c_virt_ctrl_data {
+	/* Pointer to the array of I2C devices */
+	struct i3c_i2c_device_desc *i2c_devs;
+
+	/* Number of I2C devices */
+	const size_t num_i2c_devs;
+};
+
+static struct i3c_i2c_device_desc *find_i2c_dev(const struct i2c_virt_ctrl_data *data,
+						uint8_t addr)
+{
+	int i;
+
+	for (i = 0; i < data->num_i2c_devs; i++) {
+		if (data->i2c_devs[i].addr == addr) {
+			return &data->i2c_devs[i];
+		}
+	}
+
+	return NULL;
+}
+
+static int i2c_virt_ctrl_configure(const struct device *dev, uint32_t dev_config)
+{
+	return 0;
```

Comment:
I don't know if you haven't gotten to it yet, but there should a function callback for this to the i3c controller.

Reply:
Actual configuration should be done to the I3C controller driver instance only. The virtual controller is really a intermediate layer to ease transition. If runtime configuration of the controller from the target device drivers, they really should use the I3C API.

Reply:
I don't see why it's a problem to ease transition even more. All the controllers that I've worked with keep the i3c and i2c timing in separate registers. This function could call a new api function that would only configure i2c timing within the i3c controller...

Reply:
Setting the I2C part aside, the I3C API to configure the controller is TBD, as I am not sure how many things needs to be configured at this point. There might be many options so these cannot be stuffed within a uint32_t.